### PR TITLE
TypeScript

### LINKED ISSUE #12: TypeScript
I've ported the project to TypeScript in one of my branches and that actually caught a few issues already. Should I open a PR for that or are you not interested in using TypeScript in this project? 

/cc @krisselden @stefanpenner

### PR DESCRIPTION
**Pull request by Turbo87**
This PR ports the project to TypeScript. The TypeScript sources are in `src/` now and they get compiled to `lib/` by `npm run build` which gets called by the `prepublish` hook. `npm test` is running `mocha` with the `ts-node` transpiler attached.

Resolves #12 

/cc @krisselden @stefanpenner

### CODE REVIEW

**Comment by stefanpenner** [comment]
Looks very good, 👍 . I don't have commit/merge on this repo, so we will need @krisselden 👀  as well.

**Comment by krisselden** [comment]
@Turbo87 did you consider using `npm install @types/mocha --save-dev` etc, instead of typings?

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  "compilerOptions": {
+    "module": "commonjs",
+    "target": "es5",
+    "sourceMap": true,
+    "outDir": "lib/",
+    "declaration": true
```
Comment (krisselden) [inline tsconfig.json:7]: consider adding strictNullChecks

Comment (krisselden) [inline tsconfig.json:7]: also, noImplicitAny and noImplicitReturns

**Comment by krisselden** [comment]
Though, I think @types requires `npm install typescript@next --save-dev`

File: tsconfig.json (hunk 1 above)
Comment (Turbo87) [inline tsconfig.json:7]: I got another PR already lined up based on this one and I can look into adding it there. I wouldn't like to generate too many unnecessary merge conflicts for myself if you're okay with that 😉

**Comment by Turbo87** [comment]
> @Turbo87 did you consider using npm install @types/mocha --save-dev etc, instead of typings?

I'm pretty new to TypeScript and basically just followed what I found on the web. I'll look into changing it. Feel free to merge this already though if you think it's a good first step.

**Pull request merged**